src/main/java/seedu/address/model/person/Level.java

```diff
@@ -9,16 +9,23 @@
  */
 public class Level {
 
-    public static final String MESSAGE_CONSTRAINTS = "Education level should only contain alphanumeric characters "
-            + "and spaces, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Education level should only contain the prefix [p, s, j], "
+            + "followed by the year of study eg. p2, s4, j1"
+            + "Primary: years 1-6    Secondary: years 1-5    JC: years 1-2";
     /*
-     * The first character of the education level must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
+     * The first character has to be either p, s, j.
+     * For p valid years are 1-6
+     * For s, valid years are 1-5
+     * For j, valid years are 1-2
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String VALIDATION_REGEX_PRIMARY = "[p][1-6]";
+    public static final String VALIDATION_REGEX_SECONDARY = "[s][1-5]";
+    public static final String VALIDATION_REGEX_JC = "[j][1-2]";
 
     public final String value;
 
+    public final String index;
```

Comment:
I agree with Brandon. Not sure if having both `value` and `index` is necessary. Also if we _are_ planning to have both, then maybe it is better for `toString()` and `equals()` to refer to `index` instead of `value` (in case we are planning to take in case-insensitive inputs, e.g. both p5 and P5).